### PR TITLE
(feature) Allow custom hide columns action

### PR DESCRIPTION
**Pull request by EmilianoMorghen**
Dear Team,

First and foremost, I want to express my heartfelt gratitude for the incredible work each of you has contributed. Over the past year, I've had the pleasure of utilizing this package, and I must say, I've thoroughly enjoyed it.

> TL;DR
> I'm in need of overwriting the default 'hide column' action.

## Description

Upon delving into how custom actions are rendered and handled, I've observed that they lack access to crucial data such as `columns` and event callbacks like `onColumnsChanged` or `onSearchChanged`. Providing access to this data would empower users to customize the behavior of default actions.

## Impacted Areas in Application

List general components of the application that this PR will affect:
- `MTableToolbar`
- `MTableActions`
- `MTableAction`

## Additional Notes

There are multiple approaches to implementing the overwrite of default actions' behavior, and I am open to hearing your insights on the matter.

I am eager to tackle this task with your expert guidance and oversight, ensuring that we maintain the highest quality standards that you consistently uphold.

### CODE REVIEW

File: src/components/MTableToolbar/index.js [hunk 1]
```diff
@@ -248,6 +248,10 @@ export function MTableToolbar(props) {
               props.actions &&
               props.actions.filter((a) => a.position === diplayedActions)
             }
+            columns={props.columns}
+            onColumnsChanged={(col, hidden) =>
+              props.onColumnsChanged(col, hidden)
```
Comment (Domino987) [inline src/components/MTableToolbar/index.js]: Canyou add a check if props.onColumnsChanged is not undefined?

**Review by Domino987 — CHANGES_REQUESTED** [review]
This looks good and should be a good addition, one minor change it is good to go. Thank you

File: src/components/MTableToolbar/index.js (hunk 1 above)
Comment (EmilianoMorghen) [reply src/components/MTableToolbar/index.js]: Done! Let me know if I can do anything more :)

Comment (EmilianoMorghen) [reply src/components/MTableToolbar/index.js]: @Domino987 do you have any update on this one?

**Review by Domino987 — APPROVED** [review]

**Pull request merged**